pkg/services/sqldb/provision.go

```diff
@@ -95,16 +176,96 @@ func (s *serviceManager) preProvision(
 	return dt, nil
 }
 
-func buildARMTemplateParameters(
+func (v *vmOnlyManager) preProvision(
+	_ context.Context,
+	instance service.Instance,
+	_ service.Plan,
+	_ service.Instance, //reference instance
+) (service.InstanceDetails, error) {
+	dt, ok := instance.Details.(*mssqlVMOnlyInstanceDetails)
+	if !ok {
+		return nil, errors.New(
+			"error casting instance.Details as *mssqlVMOnlyInstanceDetails",
+		)
+	}
+	dt.ARMDeploymentName = uuid.NewV4().String()
+	dt.ServerName = uuid.NewV4().String()
+	dt.AdministratorLogin = generate.NewIdentifier()
+	dt.AdministratorLoginPassword = generate.NewPassword()
+	return dt, nil
+}
+
+func (d *dbOnlyManager) preProvision(
+	_ context.Context,
+	instance service.Instance,
+	_ service.Plan,
+	referenceInstance service.Instance, //reference instance
```

Comment:
Nit: Remove the comment. The purpose of the argument is clear from its name. Let's just include such a comment where we've used `_` to express our disinterest in an argument and don't want to lose track of what that arg is for, in case we wish to use it in the future.

Reply:
Same feedback applies in the other provisioning steps used by this manager.